python/_dynet.pyx

```diff
@@ -1143,28 +1143,70 @@ cdef class ParameterCollection: # {{{
+        """Add a parameter to the ParameterCollection with a given initializer. There are different ways of specifying an initializer:
+
+        .. code-block:: python
+            
+            p = m.add_parameters((3,5), init=0)                         # Creates 3x5 matrix filled with 0 (or any other float)
+            p = m.add_parameters((3,5), init='uniform', scale=a)        # Creates 3x5 matrix initialized with U([-a,a])
+            p = m.add_parameters((3,5), init='normal', mean=a, std=b)   # Creates 3x5 matrix initialized with N(a, b)
+            p = m.add_parameters((5,5), init='identity')                # Creates 5x5 identity matrix
+            p = m.add_parameters((5,5), init='saxe')                    # Creates 5x5 orthogonal matrix (NOT SUPPORTED YET)
+            p = m.add_parameters((3,5), init='glorot')                  # Creates 3x5 matrix with glorot init
+            p = m.add_parameters((3,5), init='he')                      # Creates 3x5 matrix with he init
+            p = m.add_parameters(np.zeros((3,5)))                       # Creates 3x5 matrix from a numpy array (size is inferred)
+            p = m.add_parameters((3,5), init=dy.PyInitializer())        # Any parameter initializer
         
         Args:
-            dim (tuple): Shape of the parameter
+            dim (tuple, np.ndarray): Shape of the parameter. If only a numpy array is passed the dimension is inferred
         
         Keyword Arguments:
-            init (dynet.PyInitializer): Initializer (default: GlorotInitializer)
+            init (number, string, dynet.PyInitializer): Initializer, see description for details (default: GlorotInitializer)
             name (string)             : Optional name for this parameter (default: "")
             device (string)           : Optional device name for this parameter (default: "", default device)
+            scale (number): Scale for uniform initialization
+            mean (number): Mean for normal initialization
+            std (number): Standard deviation for normal initialization
         
         Returns:
             (dynet.Parameters): Created Parameter
         """
-        assert(isinstance(dim,(tuple,int)))
+        assert (isinstance(dim,(tuple, int, np.ndarray))), 'First argument of add_parameters should be a valid dimension or a numpy array'
+        if isinstance(dim, np.ndarray):
+            return self.parameters_from_numpy(dim, name=name, device=device)
+        elif isinstance(dim, int):
+            dim = (dim,)
         cdef CParameters p
         cdef CParameterInit *initializer
         cdef CDevice *dev
         cdef string _name = <string> name.encode("utf8")
+        cdef PyInitializer pyinit
         if init is None:
-            init = GlorotInitializer()
-        initializer = init.initializer
+            pyinit = GlorotInitializer()
+        else:
+            if isinstance(init, (int, float)):
```

Comment:
It is better that we can support use python list/tuple to initialize parameters.

```python
init = [1,2] # or (1, 2)
a = m.add_parameters((1,2), init=init)
```

Reply:
I can fix this